storage/badger/protocol_state.go

```diff
@@ -0,0 +1,180 @@
+package badger
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/order"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/badger/operation"
+	"github.com/onflow/flow-go/storage/badger/transaction"
+)
+
+// ProtocolState implements persistent storage for storing entities of protocol state.
+// Protocol state uses an embedded cache without storing capabilities(store happens on first retrieval) to avoid unnecessary
```

Comment:
I am wondering how much of a challenge it would be to actually cache the `RichProtocolStateEntry` on store. At the moment, we insert a `ProtocolStateEntry` (non-rich) -- presumably this is the reason for not storing the `RichProtocolStateEntry`. However, to me this seems to be an artifact that could indicate room for improvement in the software design:
* I think for about 95% of the blocks that we store, we need the `RichProtocolStateEntry` state later. This is because we finalize 95% of all proposed blocks, i.e. the blocks have children; I would guess we that we query the `RichProtocolStateEntry` when processing a block's child. 
* To me it would be most natural if the process evolving the identity table produced the `RichProtocolStateEntry`. The `RichProtocolStateEntry` would ideally provide methods to generate/return its condensed version.